packages/react/src/formElements/MdAutocompleteAsync.tsx

```diff
@@ -0,0 +1,236 @@
+    helpText,
+    error = false,
+    errorText,
+    prefixIcon = null,
+    onSelected,
+    onChange,
+    required,
+    displayValueAndText,
+    autoComplete = 'off',
+    ...otherProps
+}: MdAutocompleteAsyncProps) => {
+    const autocompleteId = id && id !== '' ? id : uuidv4();
+    const [open, setOpen] = useState(false);
+    const [helpOpen, setHelpOpen] = useState(false);
+    const [autocompleteValue, setAutocompleteValue] = useState('');
+    const [results, setResults] = useState<MdAutocompleteAsyncOptionProps[]>([]);
+    const [focused, setFocused] = React.useState(false);
+    const [displayValue, setDisplayValue] = React.useState(placeholder);
+
+    const onFocus = () => {
+        setFocused(true);
+        setOpen(true);
+    }
+
+    const classNames = classnames('md-autocomplete', {
+        'md-autocomplete--open': !!open,
+        'md-autocomplete--error': !!error,
+        'md-autocomplete--disabled': !!disabled,
+        'md-autocomplete--medium': size === 'medium',
+        'md-autocomplete--small': size === 'small'
+    });
+
+    const inputClassNames = classnames('md-autocomplete__input', {
+        'md-autocomplete__input--open': !!open,
+        'md-autocomplete__input--error': !!error,
+        'md-autocomplete__input--has-prefix': prefixIcon !== null && prefixIcon !== '',
+    });
+
+    const optionClass = (option: MdAutocompleteAsyncOptionProps) => {
+        return classnames('md-autocomplete__dropdown-item', {
+            'md-autocomplete__dropdown-item--selected': isSelectedOption(option)
+        });
+    }
+
+    const isSelectedOption = (option: MdAutocompleteAsyncOptionProps) => {
+        return value && value !== '' && value == option.value;
+    }
+
+    const handleOptionClick = (option: MdAutocompleteAsyncOptionProps) => {
+        onSelected(option);
+        setAutocompleteValue('');
+        setDisplayValue(isSelectedOption(option) ? '' : displayValueAndText ? option.value + ' ' + option.text : option.text);
+        setResults(isSelectedOption(option) ? [] : [option])
+        setOpen(false);
+        setFocused(false);
+    }
+
+
+    useEffect(() => {
+        console.log("autocompleteValue changed");
+        if (autocompleteValue) {
+            optionsLoader(autocompleteValue)
+                .then(result => {
+                    setResults(result);
+                    if (result && focused) {
+                        setOpen(true);
+                    }
+                });
+        }
+    }, [autocompleteValue])
+
+    useEffect(() => {
+        if(value && value != ''){
+            setDisplayValue(value);
+        }
+    }, []);
+
+
+    return (
+        <div className={classNames}>
+            <div className="md-autocomplete__label">
+                {label && label !== '' && (
+                    <label id={`md-autocomplete_label_${autocompleteId}`} htmlFor={autocompleteId}>
+                        {label}
+                        {required && <RequiredAsterisk/>}
```

Comment:
Dette kan løses ved å sette aria-required="true" på MdAutocomplete når man bruker den i et prosjekt, same nanvgi label med asteriks.
Skal vi gjøre dette burde kanskje være tenkt gjennom for MdSelect, MdAutocomplete og MdMultiselect i samme slengen.
I tillegg burde symbolet ha aria-hidden hvis det per nå blir lest opp av skjermleser.

Reply:
Hmm her vet jeg ikke om jeg skjønte helt hva du mente, gjerne utdyp 🤔 required her er for logikken av å vise og skjule RequiredAsterisk som er nederst i fila. Er aria-hidden nødvendig da, for skjermlesere leser vel ikke opp elementer som ikke eksisterer i DOMen? 

Men skal legge til aria-required på input-compobox elementet og 😄 Skulle vi gjort lignende for MdSelect, osv også? Kan ikke se noe om required på noen av de komponentene du nevner.

Reply:
> Sånn at når jeg legger aria-required="true" i mitt prosjekt så får inputen i autocompleten required. Altså trenger vi ikke en required-prop. Så kan man legge asteriksen i label. 

Her er jeg ikke helt enig. Selv om browser behandler ting som required pga aria-required, så er aria-props i prinsippet til å beskrive accessibility, ikke for å beskrive funksjonalitet. Docs for aria-required beskriver at denne bør sameksistere med `required`. Vi kan også oppleve at aria-required funker i f. eks Chromium, men ikke for Safari/andre nettlesere. Synes derfor vi bør kjøre begge om noe er required.

Reply:
Ser ikke ut som "required" legges på input-elementet uansett.

Reply:
@aurorascharff Leste docs litt nøyere. Ser at jeg tolket det feil opprinnelig. 

`Required` skal brukes for semantiske elementer som støtter dette slik som input, textarea og select. En tilrettelagt nettleser vil forklare brukeren at dette elementet er påkrevd bare basert på dette. (i.e aria-required trengs ikke for disse i tillegg til required) I tillegg kan man gjerne forklare nærmere hva elementet er om det ikke skulle være åpenbart for svaksynte via andre aria-props. 

For ikke-semantiske elementer (f. eks dersom man bruker en div i JS til å bygge funksjonaliteten) så skal aria-required brukes for å indikere required state. 

Så dersom et input-felt skal være required, så bør vi legge `required` prop på dette og evt. forklare bruken nærmere via andre aria-props.  

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-required

Reply:
Yes! Så beste løsning her blir kanskje å legge required prop og sende ned til sematisk element på input og textarea, dvs MdInput, MdAutocomplete og MdTextArea. Og bruke aria-required som jeg viste. Jeg tror det da ikke er behov for å legge asterisk i md-components, hva tenker dere?

Reply:
Jeg har snakket litt med carine om dette, hun mente at asterisk er OK for visuelle brukere og aria-required for skjermleser. Men da kan man altså legge asterisken i label, typ label="Enhet*" og aria-required="true" som sendes ned via {...otherProps}.

Reply:
Designsystemet har ingen "required" state fra før av.

Reply:
Ok, så om jeg forstod det riktig så har vi ikke med automatisk dynamisk setting av asteriks men istedenfor at det sendes inn manuelt som del av label. Og at egenspesifisert required prop fjernes slik at kun aria-required definerer om feltet er obligatorisk eller ikke. Er det korrekt? 😄

Reply:
Jeg tror det! Haha, godt tolket. Ifølge Carine Lynghaug som er god på UU. Men som Thomas sier kan vi vurdere å legge required som en prop også, men jeg tror det holder foreløpig uten. Så kan vi ta det som egen PR.